### ISSUE TITLE
BUG: Negative timestamps generated by KS4

### ISSUE THREAD

**Issue by h21ak9** [MouseLand/Kilosort#645]
### Describe the issue:

I ran Kilosort on two 256-channel datasets recorded at 40 kHz with a SiNAPS probe. While examining the results in Python, I noticed that there are a handful of spikes with negative timestamps.

### Reproduce the bug:

```python
Here is the code that I ran in Python to find the negative timestamps and their associated clusters.

import numpy as np
spike_times = np.load('spike_times.npy')
spike_clusters = np.load('spike_clusters.npy')
negative_timestamp_indexes = np.where(spike_times < 0)[0]
print(negative_timestamp_indexes.tolist())
print(spike_times[negative_timestamp_indexes])
print(spike_clusters[negative_timestamp_indexes])

The output looks like this:

[0, 1, 2, 3]
[-10 -10 -10 -9]
[51 30 24 41]
```

### Error message:

```shell
No error message.
```

### Version information:

Windows 11, Python 3.9 (anaconda)

### Context for the issue:

Downstream applications may object to loading spikes whose timestamps are negative. If negative timestamps are actually expected behavior then we could remove those spikes manually prior to loading the data in the downstream apps. But if not, then I thought this might be an issue worth raising.

**Comment by jacobpennington** [comment]
@marius10p is there any special behavior with the edge of the first batch that might make these show up erroneously? As long as the spike times are small (equivalent -0.25ms in this case) these could be correctly detected spikes that occured just before the recording started, right?

**Comment by jonahpearl** [comment]
Hi all — judging by the [4.0.7](https://github.com/MouseLand/Kilosort/releases/tag/v4.0.7) release notes, this is still an issue. I ran into this on KS4 4.0, and the issue seems reproducible across runs (I ran it twice, it happened twice in the same place). I'm happy to provide the data if it helps, I remember seeing zach say somewhere that it was tough to find data that caused this bug. Or happy to update to 4.0.7/8 and try again.
